staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -190,6 +201,12 @@ func (r *proxyHandler) removeAPIService() {
 	r.proxyRoundTripper = nil
 }
 
+func (r *proxyHandler) isLocal() bool {
+	r.lock.RLock()
```

Comment:
probably worth doing an atomic read or RWLock in the future since the read/write ratio to this is going to be phenomenally high